pkg/kubelet/eviction/helpers.go

```diff
@@ -532,15 +570,24 @@ func makeSignalObservations(summaryProvider stats.SummaryProvider) (signalObserv
 	// build an evaluation context for current eviction signals
 	result := signalObservations{}
 
-	if memory := summary.Node.Memory; memory != nil && memory.AvailableBytes != nil {
-		result[SignalMemoryAvailable] = resource.NewQuantity(int64(*memory.AvailableBytes), resource.BinarySI)
+	if memory := summary.Node.Memory; memory != nil && memory.AvailableBytes != nil && machineInfo != nil {
```

Comment:
open question for all of these, but should we only make an observation if capacity is !=0 for each of these resources?  i am worried about subtle bugs in cadvisor potentially causing disastrous outcomes.